lib/index.js

```diff
@@ -71,6 +71,15 @@ Strategy.prototype.authenticate = function (req, options) {
   if (req.query && req.query.error) {
     return this.fail(req.query.error);
   }
+
+  if (req.session.authOptions) {
```

Comment:
Can you explain a bit what's going on here? Where does `validate` come from?

Reply:
Renaming `validate` to `verifyWrapper`.  The wrapper is to avoid rewriting https://github.com/jaredhanson/passport-oauth2/blob/master/lib/strategy.js#L173-L211 and plugging directly into the verify function.

The `authOptions` are stored in the session, in order to verify the id token against those options after it is returned.